### PR TITLE
chore(web): patching up the user-testing page suite 🧩

### PR DESCRIPTION
**Pull request by jahorton**
Before we can start running user tests on KMW for this feature branch... we need to get the user testing pages up and working again.  That's what this PR is focused on.

For anyone curious and/or wanting to do a smidge of early user testing, the "KeymanWeb Test Home" link by the bot below is now actually useful for trying out modularized KMW.

------

So, I took the time to investigate each of the test pages and update them for compatibility with the newly-modularized Keyman Engine for Web.  Most pages work well at this point, though some will require further updates:
- The 'spacebar text' test page requires the `util.setOptions` API function, which is converted later
- The "Sentry Integration" test page links to a script outside of the `web/` folder, which can cause issues when being hosted for user testing.
    - localhost testing usually puts `web/` at top level, leaving said script inaccessible
    - CI builds generally don't preserve that specific script's location as an artifact.
- The `util.wait` test page's design requires a published API function on `keyman.util`.  That's doable, but it's actually not officially published on [the help site](https://help.keyman.com/developer/engine/web/17.0/reference/util/).
    - Since it's not published, I lean toward 'dropping it' from `keyman.util`... but I'm not sure that I should make that call.
    - The design is such that I'm even leaning toward dropping _the test page_ if we do drop the `keyman.util.wait` reference point.
        - Said design: the util method is only callable through a class instance that won't exist if the `useAlerts` option is disabled.
- The "beep" (error-feedback) test page - 'beep' gets handled in a later dev-branch
- There are also a few general bugs remaining for mobile device support that would stand out prominently... that also get their own dev-branch. 

I'm also a bit confused on the "Test Start of Sentence" test page.  I'm not sure what I'm supposed to do with its keyboard to test for desired behaviors... and the keyboard's original source is missing, so I can't infer from that.  FWIW, EuroLatin auto-capitalization (on the predictive-text test pages) seems intact, implying the important parts are still working... but maybe there's something else I haven't yet considered there.

------

There are a number of smaller bugs that get fixes in this PR, as well as a fair few newly-converted KMW API functions.

For the API functions, I only converted 'em if it was straightforward - which is why "beep", "spacebar text", and "util.wait" got left out of this PR.

For "smaller bugs", I only fixed them if the fixes were reasonably focused, self-contained, and directly motivated by behaviors easily seen when interacting with the test page:
- The alert and wait box had some display / CSS issues.  Some were actually longstanding but still easy to fix.
- Toggle UI button styling issue
- Iframe elements & attachment
- OSK state management when there is no active keyboard (as when "(System keyboard)" is selected on desktop devices)
- Lexical model registration
- Keyboard-documentation rendering
- The "CKEditor integration" test page came in very useful for resolving selection issues for design iframes and contenteditables and when there is no actual pre-set caret location.
    - There's an intermediate state during sequential event handling within CKEditor (so far as I can tell) that resulted in the latter case (no pre-set caret location) even when there _was_ an intended caret location.

As that's a number of smaller bugs, and I've been mercilessly hammering the build agents lately, I figured it'd be best to throw them all in together to reduce the overall load.

As there are followups that will resolve some _very_ significant issues that absolutely would turn up in user testing...

@keymanapp-test-bot skip

We _are_ getting much closer to actually specifying user tests, though.

### CODE REVIEW

**Comment by mcdurdin** [comment]
> As that's a number of smaller bugs, and I've been mercilessly hammering the build agents lately, I figured it'd be best to throw them all in together to reduce the overall load.

I'd prefer to hammer the build agents and keep each bug fix separate! From a development perspective, it makes life _much_ easier if there are later side-effects.

File: web/src/engine/element-wrappers/src/designIFrame.ts [hunk 1]
```diff
@@ -125,7 +125,7 @@ export default class DesignIFrame extends OutputTarget<{}> {
   }
 
   getDeadkeyCaret(): number {
-    return this.getTextBeforeCaret().kmwLength();
+    return (this.getTextBeforeCaret() ?? this.getText()).kmwLength();
```
Comment (mcdurdin) [inline web/src/engine/element-wrappers/src/designIFrame.ts:128]: See contentEditable.ts:106

**Review by mcdurdin — APPROVED** [review]
LGTM

File: common/web/keyboard-processor/src/keyboards/activeLayout.ts [hunk 2]
```diff
@@ -5,9 +5,7 @@ import type { KeyDistribution } from "../text/keyEvent.js";
 import type { LayoutKey, LayoutRow, LayoutLayer, LayoutFormFactor, ButtonClass } from "./defaultLayouts.js";
 import type Keyboard from "./keyboard.js";
 
-import KeyboardProcessor from "../text/keyboardProcessor.js";
-
-import { deepCopy, type DeviceSpec } from "@keymanapp/web-utils";
```
[the remark this answers is not in the document]
Comment (jahorton) [reply common/web/keyboard-processor/src/keyboards/activeLayout.ts:10]: Unfortunately, no.

https://github.com/keymanapp/keyman/blob/d59d544406c906c7313840a79aacbf03d179d0a7/common/web/keyboard-processor/src/keyboards/defaultLayouts.ts#L137-L138

This is used to generate the initial default layout structure for keyboards that don't provide one.  That data is then "filled in" based on the desktop form-factor layout spec.

File: web/src/engine/element-wrappers/src/contentEditable.ts [hunk 3]
```diff
@@ -103,12 +103,12 @@ export default class ContentEditable extends OutputTarget<{}> {
   }
 
   getDeadkeyCaret(): number {
-    return (this.getTextBeforeCaret() ?? this.getText()).kmwLength();
+    return this.getTextBeforeCaret().kmwLength();
```
Comment (mcdurdin) [inline web/src/engine/element-wrappers/src/contentEditable.ts]: This is missing the fix for getTextBeforeCaret

Comment (jahorton) [reply web/src/engine/element-wrappers/src/contentEditable.ts]: See below, in `getTextBeforeCaret` - the fix is implemented there.

Comment (mcdurdin) [reply web/src/engine/element-wrappers/src/contentEditable.ts]: Curious, when I added that comment, the fix was definitely not visible in GH -- I had to expand the source to find `getTextBeforeCaret()`. Very weird. LGTM now.

**Pull request merged**